### PR TITLE
Change

### PR DESCRIPTION
**Pull request by Sosthene00**
Add the logic necessary to add a change output in our transaction. The change label is now always checked when scanning. Output spent status now have a cleaner enum and a Mined() member.

### CODE REVIEW

File: src/spclient.rs [hunk 1]
```diff
@@ -353,18 +363,23 @@ impl SpClient {
         if let Some(deduce_from) = payer_vouts.choose(&mut rng) {
             let output = &mut psbt.unsigned_tx.output[*deduce_from as usize];
             let old_value = output.value;
-            output.value = old_value - Amount::from_sat(fee_amt);
+            output.value = old_value - Amount::from_sat(fee_amt - dust); // account for eventual dust
```
Comment (cygnet3) [inline src/spclient.rs:398]: What happens if the dust value is higher than the fee_amt required?

File: src/spclient.rs [hunk 2]
```diff
@@ -353,18 +363,23 @@ impl SpClient {
         if let Some(deduce_from) = payer_vouts.choose(&mut rng) {
```
Comment (cygnet3) [inline src/spclient.rs:395]: Later we should make it so that it will only deduct fees from the change address (if it exists)

Comment (Sosthene00) [reply src/spclient.rs]: Actually I like having the option to deduce from another outputs, even if the change (when it exists) is the obvious default.

Comment (cygnet3) [reply src/spclient.rs]: Made a new issue for this

**Pull request merged**